engines/order_management/app/services/order_management/subscriptions/validator.rb

```diff
@@ -107,7 +107,7 @@ def requested_variants_available?
       end
 
       def verify_availability_of(variant)
-        return if available_variant_ids.include? variant.id
+        return if (available_variant_ids.include? variant.id) || variant.deleted_at.present?
```

Comment:
This now reads as: "Deleted variants are available." I would imagine that you can now add deleted variants to a subscription as well, right?

While this seems to be the simplest fix, I wonder if it would make more sense to modify `requested_variants_available?`. That method checks all items:
https://github.com/openfoodfoundation/openfoodnetwork/blob/67d3bd7414fc8def66996b800444de0a02b1b378/engines/order_management/app/services/order_management/subscriptions/validator.rb#L105-L107

Instead of checking all items, it could ignore items that are marked for destruction.

We have similar logic already:
https://github.com/openfoodfoundation/openfoodnetwork/blob/67d3bd7414fc8def66996b800444de0a02b1b378/engines/order_management/app/services/order_management/subscriptions/validator.rb#L99-L103

It may be worth introducing a new method, for example:
```rb
      def persisting_line_items
        subscription_line_items.except(&:marked_for_destruction?)
      end
```

Reply:
Thanks for the help on this 👍